Firmware/FFBoard/Src/ShifterAnalog.cpp

```diff
@@ -65,58 +74,188 @@ void ShifterAnalog::readButtons(uint32_t* buf){
 			  gear=4;       // 4th gear
 		}
 
-		if(gear == 6 && (HAL_GPIO_ReadPin(rev_port,rev_pin) == GPIO_PIN_SET)){
+		if(gear == 6 && reverseButtonState){
 			gear = 7; // Reverse
 		}
 	}
+}
 
-	if(gear > 0){
-		*buf = 1 << (gear-1);
+void ShifterAnalog::updateReverseState() {
+	if (g27ShifterButtonClient) {
+		reverseButtonState = g27ShifterButtonClient->getReverseButton();
+	} else {
+		reverseButtonState = LocalButtons::readButton(reverseButtonNum);
```

Comment:
This is the part of this PR that I'm least happy with... If you have any better ideas I'd be happy to hear them!